### PR TITLE
Add media indicator to promo

### LINKED ISSUE #1793: Integrate StoryPromo with mediaIndicator into frontpage
**Is your feature request related to a problem? Please describe.**
Use a story promo component with mediaIndicator within the front page container

**Describe the solution you'd like**
Make sure that mediaIndicator displays correctly, given that the asset provided contains a media story (`MAP`) with `duration` key, converting the duration from seconds to minutes and seconds (eg. 2:15) and remembering to pass in whether the asset is audio or video 

**Testing notes**
[Tester to complete]

Dev insight: Unit tests, snaps and storybook

Ref: https://github.com/bbc/simorgh/issues/1574

### LINKED ISSUE #1940: Update story promo link
**Is your feature request related to a problem? Please describe.**
 Due to ‘text splitting’ in VoiceOver iPhone - which splits the link/h3 into 3 elements, meaning 3 swipes for each link/heading.

**Describe the solution you'd like**
Inside the current Headline Link, when the media indicator is showing, use spans to structure the text where the current headline is.

Likely highly integrated with https://github.com/bbc/simorgh/issues/1793

```jsx
<Headline> /* h3 */
  <Link href="/somelink">
    <span role="text">
      <span>Video, </span> /* media type */
      <span>Restoring a pharaoh's ancient boat</span> /* headline */
      <span datetime="PT1M33S">, 1:33</span> /* duration */
    </span>
  </Link>
</Headline>
```
**Describe alternatives you've considered** 
Create a component (Name TBC)  inside `StoryPromo` that can be used in the below manner

```jsx
<Component href='/somelink' type='video', duration='duration' headline='headline` />
```
**Testing notes**
[Tester to complete]

Dev insight: Regression on elements that use `Link` as a styled `a` tag.

**Additional context**

### PR DESCRIPTION
**Pull request by andrew-nowak**
Resolves #1793
Resolves #1940

**Overall change:** Show a media indicator on promos for media (audio or video) content

**Code changes:**

- Bring in psammead-media-indicator
- Add a duration formatter to format moment durations to numbers separated by colons eg. '1:02:03'
  - Challenge for reviewers: are there edge cases that I've missed in the unit tests?
  - Alternative was to use https://www.npmjs.com/package/moment-duration-format, but https://bundlephobia.com/result?p=moment-duration-format@2.3.2 :(
- Put formatted durations (ISO and the above style) into ~~a MediaIndicator~~ some visually hidden text near the headline #1940 if available in the data.
- Pass translations of 'audio' and 'video' (taken from the live site) into offscreen text ~~in the media indicator~~ near the headline #1940 

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/StoryPromo/index.jsx [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
 } from '@bbc/psammead-story-promo';
 import Timestamp from '@bbc/psammead-timestamp-container';
+import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
 import { storyItem } from '../../models/propTypes/storyItem';
 import StoryPromoFigure from './Figure';
 import { ServiceContext } from '../../contexts/ServiceContext';
 import deepGet from '../../lib/json/deepGet';
+import formatDuration from '../../lib/utilities/formatDuration';
+
+const buildMediaIndicator = (item, mediaTranslations) => {
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: Can these go in a new file or 2? its getting pretty big, probably would be nice to unit test them individually too maybe?

File: src/app/containers/StoryPromo/index.jsx [hunk 2]
```diff
@@ -1,18 +1,87 @@
 import React, { Fragment, useContext } from 'react';
+import moment from 'moment';
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: You can import moment from `moment-timezone` here, means we dont have to have 2 moment packages in our package json :)

File: src/app/containers/StoryPromo/index.jsx [hunk 3]
```diff
@@ -1,18 +1,87 @@
...
+  const duration = moment.duration(rawDuration, 'seconds');
+  const durationString = formatDuration(duration);
+  const isoDuration = duration.toISOString();
+
+  return (
+    // role="text" is required to correct a text splitting bug on iOS VoiceOver.
+    // eslint-disable-next-line jsx-a11y/aria-role
+    <span role="text">
+      <VisuallyHiddenText>{mediaTranslations[type]}, </VisuallyHiddenText>
+      <span>{headline}</span>
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: If its not a media promo, do we not want to just return text here? one for @greenc05 but at a guess, we shouldn't wrap it in a span if we dont need to

**Review by dr3 — CHANGES_REQUESTED** [review]
Great work, just a couple things

File: src/app/containers/StoryPromo/index.jsx (hunk 3 above)
Comment (andrew-nowak) [reply src/app/containers/StoryPromo/index.jsx]: see https://github.com/bbc/simorgh/pull/1918/files/0b9a83cbcc14b637ca05edb0386c758f47a82b03#diff-5a217b9ae93005d18abe435f36b7ee9aR53

Comment (dr3) [reply src/app/containers/StoryPromo/index.jsx]: Hahah whoops missed that 🙈

File: src/app/containers/StoryPromo/index.jsx [hunk 4]
```diff
@@ -1,18 +1,87 @@
...
+      {rawDuration && (
+        <span>
```
Comment (dr3) [inline src/app/containers/StoryPromo/index.jsx]: VisuallyHiddenText is a span, so doesn't need to be wrapped in one :)

**Review by dr3 — COMMENTED** [review]
Annoyingly because of https://github.com/bbc/psammead/blob/latest/packages/components/psammead-media-indicator/src/index.jsx#L35 the media indicator doesn't show the duration value without offscreen text, drats

File: src/app/containers/StoryPromo/LinkContents.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    // role="text" is required to correct a text splitting bug on iOS VoiceOver.
+    // eslint-disable-next-line jsx-a11y/aria-role
+    <span role="text">
+      <VisuallyHiddenText>{mediaTranslations[type]}, </VisuallyHiddenText>
+      <span>{headline}</span>
+      {rawDuration && (
+        <span>
+          <VisuallyHiddenText>
+            , <time dateTime={isoDuration}>{durationString}</time>
```
Comment (andrew-nowak) [inline src/app/containers/StoryPromo/LinkContents.jsx]: The promo headline shouldn't use a `<time>` element. (Removes the need for a isoDuration)

**Comment by andrew-nowak** [comment]
~~blocked on bbc/psammead#644~~

File: src/app/containers/StoryPromo/LinkContents.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
+
+import { ServiceContext } from '../../contexts/ServiceContext';
+import deepGet from '../../lib/utilities/deepGet';
+import formatDuration from '../../lib/utilities/formatDuration';
+import { storyItem } from '../../models/propTypes/storyItem';
+
+const LinkContents = ({ item }) => {
+  const {
+    translations: { media: mediaTranslations },
```
Comment (dr3) [inline src/app/containers/StoryPromo/LinkContents.jsx:14]: Will this blow up if `mediaTranslations` is missing? Ive not done much nested destructuring before :P

Comment (dr3) [reply src/app/containers/StoryPromo/LinkContents.jsx:14]: Ah it does, but if translations is null, then were screwed so LGTM

**Review by dr3 — APPROVED** [review]

File: src/app/lib/utilities/formatDuration.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable prefer-template */
+
+const lpad = s => s.toString().padStart(2, '0');
```
Comment (dr3) [inline src/app/lib/utilities/formatDuration.js:3]: Our babel doesn't polyfil `padStart` :O

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/lib/utilities/formatDuration.js (hunk 7 above)
Comment (andrew-nowak) [reply src/app/lib/utilities/formatDuration.js:3]: We checked again and it *was* polyfilling `padStart`.

False alarm

**Review by dr3 — APPROVED** [review]
Really really nice PR, Great job 👍

File: src/app/containers/StoryPromo/LinkContents.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  if (!isMedia) {
+    return headline;
+  }
+
+  const type = deepGet(['media', 'format'], item);
+
+  // Always gets the first version. Smarter logic may be needed in the future.
+  const rawDuration = deepGet(['media', 'versions', 0, 'duration'], item);
+
+  // hilariously, this works. according to moment, null seconds == 0 seconds!
```
Comment (j-pendlebury) [inline src/app/containers/StoryPromo/LinkContents.jsx:29]: I enjoy reading your comments 😄

**Review by j-pendlebury — APPROVED** [review]
LGTM 👍

File: src/app/containers/StoryPromo/MediaIndicator.test.jsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    versions: [
+      {
+        duration: 5600,
+      },
+    ],
+  },
+};
+
+const nonMediaItem = {
+  cpsType: '🤔',
```
Comment (DenisHdz) [inline src/app/containers/StoryPromo/MediaIndicator.test.jsx:25]: Nice PR 🧐

**Review by DenisHdz — APPROVED** [review]

**Comment by david-boydell** [comment]
Looks good, ready for merge
